### PR TITLE
frontend: add yaml tags to UDP and HTTP configs

### LINKED ISSUE #228: HTTP and UDP configs need YAML tags

### PR DESCRIPTION
**Pull request by jzelinskie**
This change also adds the missing fields to the example config.

Fixes #228.

### CODE REVIEW

File: example_config.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ chihaya:
   udp:
     addr: 0.0.0.0:6881
     allow_ip_spoofing: false
+    max_clock_skew: 10s
+    private_key: |
+      paste a random string here that will be used to hmac connection IDs
```
Comment (mrd0ll4r) [inline example_config.yaml]: Why is this on two lines and why is the second line not a comment? YAML specific?

Comment (jzelinskie) [inline example_config.yaml]: if you need to insert multi-line strings in YAML you can use this pipe and as long as the stuff is properly indented, it works

i.e.

```yaml
private_key: |
  MIIEogIBAAKCAQEArxXrwqYndJlO8FOk3Jr7WzWkHeyFD8dg0wtkSSo10Nz8iDWA
  /LSL71iwUKxP29zGk8gHoiCXKFuBfRrucBk76RN0S2DP/TwsVh8cZ2G9YE/S6wsi
  ZDvPqTMBpYd3q328bH+b4RBgxDAjGInK7sltkkIN7V8pyyITPzLIz13oKmfC7LhJ
  nSmEiO9THQXMEyXJYGA/IFxPky5zhYkWYgkmevKXCFtEm3R+sfJTijFmWHw1g7dB
  TJ4UC536LQB1aZIz95IpZfSHms1BPx9dgSQnty7zdRXjoC1WRRoQUofToap7LkmZ
  3VC4q+WuDcHlharblT4XphvP97iCD/RK4v1T/wIDAQABAoIBAAKhi4hCc0UAdBwO
  ww32SEEFISu6nP9GiZEc/TX6QT0bwlxmTW4UD9GaHWrWRAxI7jT/TwS5L6b8Gl81
  EzsyWIuhRAtJRPwzC3QvMH/p4Jd8wO3Q84SFFOhKgMOqMaVX6jJz3SBN3N5l3h56
  x5wSAShZqhg/PgNe8OIaExlxTvwAH753l5lGQNvc5m9uWgwSBnrxLHf1YWGXn9YE
  GdbjC3SX4sNl6kC3luwEDEhD61vKJVrUPSVUwXZNnFB4s3RNSxXhr3rqYOpQ48zH
  UbHgzp3oJOxa9QlbSUw7g2aABK16bVJZrzQVOaNUmSaTTRNs+aoacKhfKhxNloXJ
  7d0zJfkCgYEAt+F1USt5Q13eDhzTLaRZZQbJpMbsywfEDqXwjG1L+eVYFd7/Cpli
  YNqhau5umRha+bZ7K4soWtlTzSRbU0qClHDsoJZwW5Mce2aTS1wa3lUFG0zvDTzP
  PsDJOfVz/JbiRBgSKN/pwl8JIDyXhusvJ6NMrQ1rP92mO6TOhuHxc9kCgYEA88Fk
  1SBohOV3LRYsxp04/c+mF7y6xPlerASRQTf21fHARtxT3oRt+FayF8sPtmhrc/St
  pJ6F79sPZfuv8LAsrehFb1py7gbJvZISYaLC4wpdcZwS5LCo6kgwLhc3qnaAQ1yo
  zj1d2Zr/3iMwnLbNvBHNWLWW63rkRGoTtAcFl5cCgYA/wgkvr+dIGJAaTJxcwxPM
  F+JrJB6O945fleImju89Iuy8CKpePnkCJEy8CxiGNQ+2bjOfEY1Ykuqfwu93nepR
  Fia2N/OVjRjfIN4Jz1HxrYEryi1JZcBmfepIEne5KYgjHEWPrrQbFWDQsLHJslqT
  X+gCSRja+DvfinGiC03xYQKBgCxCpYh9hfz7nZQ2PpMh4UNU/Re6UT+K+EHL4oQx
  IUTfzMYRI7l32nAJbfUcDhJVC43Wk+EcV72hzT37YYsWskYxssnxZoUJ/n/JkkTg
  RPrs2uLSMCwOV92Slg0IcjclpUyV+qDYS75DgspXIXCTUPAhW2QMNQc1HVW+3S0H
  SWCpAoGAL/MW2ybBfssTaGsTFRuxHI/BYKM/lGnF8nrfKjrDriXA7MKTnyewIBH+
  jpo8VvBlw1JEgJfEYHJlKXOWgioXh20+aMT6yMiYR0kQ437BdlqIqVFEt4H8ecjL
  en0jKHpmF5DE/9TSzgCC5tmT7OrALOvqjDw7ala6IetJZAAgAbs=

Comment (mrd0ll4r) [inline example_config.yaml]: cool! I didn't know about that.

Comment (jzelinskie) [inline example_config.yaml]: yea we use it a lot for cloundconfigs: https://coreos.com/os/docs/latest/cloud-config.html#writefiles

**Pull request merged**